ecogvis/functions/subFunctions.py

```diff
@@ -747,28 +769,36 @@ def BadChannelDel(self, ch_list):
         ch_list : list of integers
             List of indices of channels to be un-marked as 'bad'.
         """
+        # Update list of bad electrodes ids
         for ch in ch_list:
-            if ch in self.badChannels:
-                self.badChannels.remove(ch)
+            if ch in self.bad_channels_ids:
+                self.bad_channels_ids.remove(ch)
+
+        # List of electrodes IDs
+        elecs_ids = list(self.nwb.electrodes.id[:])
+        is_bad_list = [False] * len(elecs_ids)
+        for i, id in enumerate(elecs_ids):
+            if id in self.bad_channels_ids:
+                is_bad_list[i] = True
+
+        if 'bad' not in self.nwb.electrodes:
+            self.nwb.add_electrode_column(
+                name='bad',
+                description='electrode identified as too noisy',
+                data=is_bad_list,
+            )
+        else:
+            self.nwb.electrodes['bad'].data[:] = is_bad_list
+
+        # Refresh screen
         self.refreshScreen()
 
     def BadChannelSave(self):
         """Saves list of bad channels in current NWB file."""
         buttonReply = QMessageBox.question(None, ' ', 'Save Bad Channels on current NWB file?',
                                            QMessageBox.No | QMessageBox.Yes)
         if buttonReply == QMessageBox.Yes:
-            # Modify current list of bad channels
-            aux = [False] * self.nChTotal
-            for ind in self.badChannels:
-                aux[ind] = True
-            if 'bad' not in self.nwb.electrodes:
-                self.nwb.add_electrode_column(
-                    name='bad',
-                    description='electrode identified as too noisy',
-                    data=aux,
-                )
-            else:
-                self.nwb.electrodes['bad'].data[:] = aux
+            pass
```

Comment:
What is going on here? This button does nothing? Then why have it?

Reply:
fixed with 2f99a2c